contributors/design-proposals/network/service-topology-key.md

```diff
@@ -0,0 +1,103 @@
+# topology aware routing of services
+
+Author: @m1093782566
+
+## Objective
+
+Provide a generic way to provide the local-only service route, say "topology aware routing of service". Local means "same topology level", e.g. same node, same rack, same failure zone, same failure region, same cloud provider etc. To clarify, this proposal is intended to resolve the user case that user always want to visit a local service's backend even though for some reason local service's backend is not available.
+
+## GOAL
+
+A generic way to support various topology levels, regarless whatever the topology key is.
+
+## Non-goal
+
```

Comment:
non-goal : to scope or spec the changes required to the rest of the system (e.g. scheduler spreading) to implement this sort of topology guarantee.

Reply:
Got it. Thanks!

Reply:
Added now.